src/jit/lsra.cpp

```diff
@@ -7551,6 +7660,15 @@ void LinearScan::allocateRegisters()
             currentInterval->physReg               = assignedRegister;
             regsToFree &= ~assignedRegBit; // we'll set it again later if it's dead
 
+#ifdef _TARGET_ARM_
+            // Update overlapping floating point register for TYP_DOUBLE
+            if (currentInterval->registerType == TYP_DOUBLE)
+            {
+                assert(isFloatRegType(currentInterval->assignedReg->registerType));
+                regsToFree &= ~genRegMask(REG_NEXT(assignedRegister));
+            }
+#endif
+
             // If this interval is dead, free the register.
```

Comment:
If you kept only the valid half of a `TYP_DOUBLE` register in the sets of registers to free, then this would not be required.